src/Transport/Topology/TopologyCreator.cs

```diff
@@ -51,5 +51,36 @@ public TopologyCreator(ITransportPartsContainer container, IManageNamespaceManag
                 }
             }
         }
+
+        public async Task TearDownSubscription(TopologySection topologySection)
```

Comment:
Why is the delete subscription contained in this PR? I suppose those changes are unrelated to this PR?